### ISSUE TITLE
[RFC] Swap Posthtml for Reshape for HTML transforms

### ISSUE THREAD

**Issue by jssee** [parcel-bundler/parcel#310]
Reshape allows us to do all the same things PostHTML does but offers a much sturdier foundation, meaning less issues come build time. 

Unfortunately PostHTML has many plugins that are missing tests or just downright don't work. The same can be said for the project core (big test gaps and low coverage) and it doesn't seem to have very active maintainers. 

Reshape was created to address these specific flaws in PostHTML and while it doesn't have as many plugins as PostHTML does _right now_ it does offer a lot more stability, and the fundamentals are similar enough to PostHTML that it would be easy enough to swap the two, and even fill whatever gaps in the plugin ecosystem exist between the two. Not to mention the project has great active maintainers (the creator was also an active contributor of PostHTML in the early days).

more info about Reshape here: https://reshape.ml

**Comment by Ty3uK** [comment]
You can write your own asset and use Reshape in `transform` method 🤔

**Comment by Ty3uK** [comment]
Extend from `HTMLAsset` and change only `transform` method implementation

**Comment by brandon93s** [comment]
Would changing to Reshape solve a specific issue that currently plagues `parcel`?

**Comment by jssee** [comment]
My suggestion is more about preventing issues in the future than problems it fixes now.

**Comment by voischev** [comment]
> Unfortunately PostHTML has many plugins that are missing tests or just downright don't work. The same can be said for the project core (big test gaps and low coverage) and it doesn't seem to have very active maintainers.

💩 Bullshit. 

- https://github.com/reshape/reshape/issues/54
- https://coveralls.io/github/posthtml/posthtml 
- https://github.com/posthtml/posthtml/commits/master 12 days ago
- https://github.com/reshape/reshape/commits/master 20 Jul
and
- http://www.npmtrends.com/posthtml-vs-reshape

PostHTML has many users, there is a few active issue

PostHTML welcome PR 👋!

**Comment by GitScrum** [comment]
## [benchmark](https://github.com/fb55/htmlparser2#performance)
Reashe use -  parser5 
PostHTML use - htmlparser

**Comment by jescalan** [comment]
Hi friends! I don't want this to devolve into any sort of aggression here. I am the author of reshape, and it was originally created to take care of several key features I was missing from posthtml. Specifically, more thorough testing across all core repos and plugins, the ability to produce functions as output in addition to strings (which enables use for client-side templates), and more accurate error tracking using location data. I have also made a decent number of contributions to posthtml, and not only have nothing against the project, but think that it's great.

I know that, with the possible exception of adding the ability to output a function, posthtml maintainers have been working hard on filling in these gaps. And I know they have been working *especially* hard since parcel was launched. That being said, I'd like to clarify a couple things:

1. While the project wasn't super active for a good period of time, since its inclusion in parcel, it has become much more active, which is fantastic. It seems like more tests have been added, although there's [still some ways to go](https://coveralls.io/github/posthtml). It's abundantly clear just from the comments and reactions here that these guys are *very* invested in posthtml, and it seems like its inclusion in parcel was not only a positive thing for them but a jolt of life to the project, which is good for everyone involved.
2. The "benchmark" given above is not only misleading, but incorrect. It is a benchmark of the html parsers that are used by the core of either project, not the actual projects in question. For context, I chose to use parse5 for reshape because it gives quality location tracing so that clear error messages can be thrown, and because it's exceptionally good at producing spec-compliant html, as its maintained by whatwg members.

I think overall what @brandon93s said is the way to go. If there is a specific issued caused by posthtml that could be solved by swapping it for reshape, and that the maintainers are unwilling to work out, it might be worth thinking about. It seems like the maintainers are committed to this project though, and I'm sure they will be able to make it work.

All that being said, I'd also like to add some big congrats both to the maintainers and contributors of both parcel and posthtml for the hard work they have done and the positive impact they have made on the open source community 👬
